spec/components/email/receiver_spec.rb

```diff
@@ -147,17 +147,29 @@ def create_post_reply_key(value)
         expect(IncomingEmail.last.is_bounce).to eq(true)
       end
 
-      it "when bounce with verp" do
-        SiteSetting.reply_by_email_address = "foo+%{reply_key}@discourse.org"
-        bounce_key = "14b08c855160d67f2e0c2f8ef36e251e"
-        create_post_reply_key(bounce_key)
-        Fabricate(:email_log, to_address: email_address, user: user2, bounce_key: bounce_key, post: post)
+      context "when bounce with verp" do
+        let(:bounce_key) { "14b08c855160d67f2e0c2f8ef36e251e" }
 
-        expect { process(:hard_bounce_via_verp) }.to raise_error(Email::Receiver::BouncedEmailError)
-        post = Post.last
-        expect(post.whisper?).to eq(true)
-        expect(post.raw).to eq(I18n.t("system_messages.email_bounced", email: email_address, raw: "Your email bounced").strip)
-        expect(IncomingEmail.last.is_bounce).to eq(true)
+        before do
+          SiteSetting.reply_by_email_address = "foo+%{reply_key}@discourse.org"
+          create_post_reply_key(bounce_key)
+          Fabricate(:email_log, to_address: email_address, user: user2, bounce_key: bounce_key, post: post)
+        end
+
+        it "creates a post with the bounce error" do
+          expect { process(:hard_bounce_via_verp) }.to raise_error(Email::Receiver::BouncedEmailError)
+          post = Post.last
+          expect(post.whisper?).to eq(true)
+          expect(post.raw).to eq(I18n.t("system_messages.email_bounced", email: email_address, raw: "Your email bounced").strip)
+          expect(IncomingEmail.last.is_bounce).to eq(true)
+        end
+
+        it "updates the email log with the bounce error message" do
+          expect { process(:hard_bounce_via_verp) }.to raise_error(Email::Receiver::BouncedEmailError)
+          email_log = EmailLog.find_by(bounce_key: bounce_key)
+          expect(email_log.bounced).to eq(true)
+          expect(email_log.bounce_error_code).to eq("5.1.1")
```

Comment:
Hmm sorry it's not obvious to me where this "5.1.1" comes from..

Reply:
Nevermind found it.

Reply:
It's from the `hard_bounce_via_verp` email fixture, with our email specs the `process(:X)` code is shorthand to find one of these fixtures and run it through the email receiver flow:

https://github.com/discourse/discourse/blob/8aec4dc16b2f6c0b38c19591cc621249d0fab1e2/spec/fixtures/emails/hard_bounce_via_verp.eml#L18-L26